pkgs/native_assets_cli/lib/src/model/build_config.dart

```diff
@@ -80,6 +83,44 @@ class BuildConfig {
     return BuildConfig.fromConfig(config);
   }
 
+  /// Constructs a unique [outDir] for a [BuildConfig] based on the fields
+  /// of a buildconfig that influence the build.
+  ///
+  /// In particular, it only takes the package name from [packageRoot],
+  /// so that the hash is equal across checkouts and ignores [outDir] itself.
+  static String outDirName({
```

Comment:
Instead of `outDirName` we only want a hash identifying this configuration. So maybe `outDirName` -> `checksum` or something like that.

If the `Config config` (aka json) would contain `CCompilerConfig`, `LinkModePreference`, `Target` and all other configuration, then it would boil down to

```
class BuildConfig {
  final Config config;

  String get checksum => sha256(json.encode(config.asJson));
}
```

Simple :smile:

Reply:
`asJson` -> `toYaml` does the same thing, but it includes `outDir`.

What bites us here is the "atomic-ness" of a config. You can instantiate it only if it is complete and valid (e.g. the CCompilerConfig->cc needs to exist, and outDir needs to be set, etc.)

We rely on only having valid `Config` objects floating around in the heap in other places in the system, so that we don't have to check anything when reading from a `Config`.

> `checksum`

Wouldn't that be suggesting _all_ fields? e.g. have similar behavior to hashCode. Including `outDir`?